[#53] Team select dropdown - A

kammanz opened the pull request:
Refactored Player's list

What's completed:

- Refactored conditional code for sixers urlName
- Added team color property to player object, making it easier to render team color without bugs
- Added dynamic team color to player card background (on select), and line
- Fixed bug: when player card was selected, it retained its selected status in memory even after switching to another team
- Resized placeholder image so it's closer to design

Commit:
[#54] minor fix

Commit:
[#54] Changed player card underline color to team color

Commit:
[#54] minor fix

Commit:
[#54] minor fix

Commit:
[#54] Fixed merge conflicts

Commit:
[#54] Fixed merge conflicts

Commit:
[#54] Resized placeholder image

Commit:
[#53] Minor fixes

Commit:
[#53] Added team background color to player card on select

Commit:
Merge branch 'RAP-53__Team-select-refactor--header' into RAP-53__Team-select-refactor--A

Commit:
[#53] Deleted extraneous getPlayers action

Commit:
[#53] Refactored code for conditional sixers teamUrlName

Commit:
[#53] Fixed console error, added teamId property to card component so image renders

Commit:
[#53] Refactor, player objects now have a teamColor property

Commit:
[#53] Refactor, player now loses his selected status when user changes teams

kammanz commented on src/actions/actions.js:
```diff
@@ -10,26 +10,22 @@ export const getTeams = () => async dispatch => {
 
     const raptorsRosterResponse = await dataNbaNet.get('/json/cms/noseason/team/raptors/roster.json');
-    dispatch({ type: 'GET_PLAYERS', payload: raptorsRosterResponse.data.sports_content.roster.players.player });
+    const raptorsRoster = raptorsRosterResponse.data.sports_content.roster.players.player.map((player) => {
```
There was a bug; the team color would linger for a second even after the user switches teams.  So I added a property to the player object, "teamColor," which made it easier to access the team color and fixed the bug.

Tumne replied:
Duly noted, just write a small comment in your code about it.
The linger bug will disappear once we do the loading states issue here: https://github.com/kammanz/raptors-app/issues/39

kammanz commented on src/actions/actions.js:
```diff
@@ -10,26 +10,22 @@ export const getTeams = () => async dispatch => {
+    const teamRoster = teamRosterResponse.data.sports_content.roster.players.player.map((player) => {
+        return { ...player, teamColor: team.primaryColor };
+    });
+    dispatch({ type: 'GET_PLAYERS', payload: teamRoster });
```
getPlayers was extraneous so I deleted it.

Commit:
[#53] minor fixes

Commit:
[#53] Fixed player backdrop

Commit:
[#53] Reset placeholder to default on team select

Commit:
[#53] Added team color background for section titles and underline

Commit:
[#53] Added dynamic padding top for table data cells

Commit:
[#53] minor fixes

Tumne commented on src/components/players/_shared/card.js:
```diff
@@ -4,7 +4,7 @@ import placeholderImg from '../../../assets/imgs/placeholder.png';
 import styles from './card.module.scss';
 
-const Card = ({ player }) => {
+const Card = ({ player, playerTeam }) => {
```
Technically this should be called `playerTeamId` as that is what you're sending.

Tumne commented on src/components/players/_shared/card.js:
```diff
@@ -20,7 +20,7 @@ const Card = ({ player }) => {
             <div className={styles.imageContainer}>
                 <img
-                    src={`https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/1610612761/2019/260x190/${person_id}.png`} alt="player headshot"
+                    src={`https://ak-static.cms.nba.com/wp-content/uploads/headshots/nba/${playerTeam}/2019/260x190/${person_id}.png`} alt="player headshot"
```
same here

Tumne commented on src/components/players/details/index.js:
```diff
@@ -59,7 +59,7 @@ const Details = ({ player, teams, games }) => {
     return (
         <div className={styles.container}>
-            <Card player={player} />
+            <Card player={player} playerTeam={selectedTeam.teamId} />
```
same here: `playerTeamId`

Tumne commented on src/components/players/list/index.js:
```diff
@@ -14,18 +14,22 @@ class List extends React.Component {
+    componentDidUpdate(prevProps) {
+        if (prevProps.selectedTeam !== this.props.selectedTeam) {
+            this.setState({ selectedId: null });
+        };
```
👍

Tumne commented on src/components/players/list/index.js:
```diff
@@ -39,14 +43,14 @@ class List extends React.Component {
-                    <div className={styles.detailsContainer}>
-                        <div className={styles.number}>{player.jersey_number}</div>
-                        <div className={styles.details}>
+                    <div style={{ borderBottom: `2px solid ${teamColor}`}} className={styles.imageLine} />
```
You can just write this as (also notice no need for string literals):
```
 <div style={{ borderColor: teamColor }} className={styles.imageLine} />
```

and then in `.imageLine` css just do:

```
  .imageLine {
    border-bottom: 2px solid transparent;
    width: 100%;
    margin-top: 4px;
  }
```

Tumne commented on src/components/players/list/index.js:
```diff
@@ -39,14 +43,14 @@ class List extends React.Component {
-                        <div className={styles.number}>{player.jersey_number}</div>
-                        <div className={styles.details}>
+                    <div style={{ borderBottom: `2px solid ${teamColor}`}} className={styles.imageLine} />
+                    <div style={isSelected ? { backgroundColor: `${teamColor}`}: null} className={styles.detailsContainer}>
```
Should be simplified to:

```
<div style={{ backgroundColor: isSelected && teamColor }} className={styles.detailsContainer}>
```

kammanz replied:
cool. next level.

Tumne commented on src/components/players/list/index.js:
```diff
@@ -39,14 +43,14 @@ class List extends React.Component {
-                        <div className={styles.details}>
+                    <div style={{ borderBottom: `2px solid ${teamColor}`}} className={styles.imageLine} />
+                    <div style={isSelected ? { backgroundColor: `${teamColor}`}: null} className={styles.detailsContainer}>
+                        <div style={isSelected ? { backgroundColor: `${teamColor}`}: null} className={styles.number}>{player.jersey_number}</div>
```
no need for this style attribute here:
```
<div className={styles.number}>{player.jersey_number}</div>
```

Tumne commented on src/components/players/list/index.js:
```diff
@@ -39,14 +43,14 @@ class List extends React.Component {
+                    <div style={{ borderBottom: `2px solid ${teamColor}`}} className={styles.imageLine} />
+                    <div style={isSelected ? { backgroundColor: `${teamColor}`}: null} className={styles.detailsContainer}>
+                        <div style={isSelected ? { backgroundColor: `${teamColor}`}: null} className={styles.number}>{player.jersey_number}</div>
+                        <div style={isSelected ? { backgroundColor: `${teamColor}`}: null} className={styles.details}>
```
Same, no need for style attribute:

```
<div className={styles.details}>
```

Tumne commented on src/actions/actions.js:
```diff
@@ -10,26 +10,22 @@ export const getTeams = () => async dispatch => {
-export const getPlayers = teamName => async dispatch => {
-    const response = await dataNbaNet.get(`/json/cms/noseason/team/${teamName}/roster.json`);
-    dispatch({ type: 'GET_PLAYERS', payload: response.data.sports_content.roster.players.player });
+    const teamUrlName = (team.teamId === '1610612755' ? 'sixers' : team.ttsName.trim().split(' ').pop().toLowerCase());
```
Better to have a discussion about enums. The short of it is that instead of writing strings throughout our app, it's better to put these strings in an object and reference them.

This would be better discussed over a hangout or in person.

Tumne reviewed (changes requested):
Looks good, some changes. 

In addition, I'm thinking that we should include enums for constants like colors and team related values. If you check the enums/enums.js file you'll see what I mean. We'll discuss this when you have the chance.

Commit:
[#53] minor fixes

Commit:
[#53] minor fixes

Commit:
Merge branch 'RAP-53__Team-select-refactor--header' into RAP-53__Team-select-refactor--A

Commit:
[#53] merging in team-select-refactor--A

Commit:
[#53] minor changes

Commit:
[#53] Added sixers enum obj, and changed strings to enums in actions.js

kammanz commented:
Ready for review. 

Changes made:

- Added url name property to raptors enum
- Created sixers enum
- Changed strings to enums in actions.js, 
- Changed raptor variable names to 'default' in actions.js

Tumne commented on src/actions/actions.js:
```diff
@@ -1,35 +1,32 @@
 import dataNbaNet from '../apis/dataNbaNet';
+import { SIXERS_TEAM, DEFAULT_TEAM } from '../enums/enums';
```
Good to see these as enums. There can be a simplification, see my enum/enum.js comment

Tumne commented on src/actions/actions.js:
```diff
@@ -1,35 +1,32 @@
-
-    const raptorsRosterResponse = await dataNbaNet.get('/json/cms/noseason/team/raptors/roster.json');
-    dispatch({ type: 'GET_PLAYERS', payload: raptorsRosterResponse.data.sports_content.roster.players.player });
+    const defaultTeam = nbaTeamsArray.find(team => team.teamId == DEFAULT_TEAM.TEAM_ID); 
```
I understand why you did `==` instead of `===`. Because `team.teamId` is a string and the enum is a number. But it's always better to use `===`. So you can just change the enum value to a string:

In enum.js, instead `TEAM_ID: 1610612761,` becomes `TEAM_ID: '1610612761',`

Tumne commented on src/actions/actions.js:
```diff
@@ -1,35 +1,32 @@
-export const getPlayers = teamName => async dispatch => {
-    const response = await dataNbaNet.get(`/json/cms/noseason/team/${teamName}/roster.json`);
-    dispatch({ type: 'GET_PLAYERS', payload: response.data.sports_content.roster.players.player });
+    const teamUrlName = (team.teamId == SIXERS_TEAM.TEAM_ID ? SIXERS_TEAM.URL_NAME : team.ttsName.trim().split(' ').pop().toLowerCase());
```
same here, the enum for sixers should be a string. And then use `===`. 
It's a little annoying that the id that comes from our api is a string, but whatevs

Tumne commented on src/actions/actions.js:
```diff
@@ -1,35 +1,32 @@
 
 export const getTeams = () => async dispatch => {
     const allTeamsResponse = await dataNbaNet.get('/prod/2019/teams_config.json');
     const nbaTeamsArray = Object.values(allTeamsResponse.data.teams.config).filter(team => team.ttsName);
```
Let's just name it to `nbaTeams` instead of `nbaTeamsArray`. The plural on variables denotes that it is an array.

Tumne commented on src/actions/actions.js:
```diff
@@ -1,35 +1,32 @@
+    const defaultRoster = defaultRosterResponse.data.sports_content.roster.players.player.map((player) => {
+        return { ...player, teamColor: defaultTeam.primaryColor };
+    });
+    dispatch({ type: 'GET_PLAYERS', payload: defaultRoster });
```
From `line 10` to `line 17` is redundant code. lets replace it with:
```
    dispatch(getSelectedTeam(defaultTeam));
```

FYI, this is how you call an action within an action.

kammanz replied:
Nice.

Tumne commented on src/actions/actions.js:
```diff
@@ -48,6 +45,6 @@ export const getTeamRosters = () => async dispatch => {
-    dispatch({ type: 'GET_PLAYERS', payload: raptorsRosterResponse.data.sports_content.roster.players.player });
+    const defaultRosterResponse = await dataNbaNet.get(`/json/cms/noseason/team/${DEFAULT_TEAM.URL_NAME}/roster.json`);
+    dispatch({ type: 'GET_PLAYERS', payload: defaultRosterResponse.data.sports_content.roster.players.player });
 }; 
```
I don't see `export const getTeamRosters` being used anywhere in the code. let's remove this code `line 45` to `line 53`

Tumne commented on src/enums/enums.js:
```diff
@@ -14,5 +14,16 @@ export const COLOR = Object.freeze({
+export const SIXERS_TEAM = Object.freeze({
+    URL_NAME: 'sixers',
+    TEAM_ID: 1610612755,
+});
```
This works, but it can be improved. So this file should look like so:

```
/**
 * Enum for common colors.
 * @readonly
 * @enum {string}
 */
export const COLORS = Object.freeze({
    GREY: '#F5F5F5',
});

/**
 * Enum for teams object.
 * @readonly
 * @enum {{ TRI_CODE: string, NAME: string, ID: string }}
 */
export const TEAMS = Object.freeze({
    TOR: {
      TRI_CODE: 'TOR',
      NAME: 'raptors',
      ID: '1610612761',
    },
    PHI: {
      TRI_CODE: 'PHI',
      NAME: 'sixers',
      ID: '1610612755',
    },
});
```

Notice that I changed `COLOR` to `COLORS`
Moreso, now when you need to use these enums, for example if you want to use these values you call them like so:

```
import { TEAMS } from '../../enums/enums'

...

TEAMS.TOR.TRI_CODE //  'TOR'
TEAMS.TOR.ID //  '1610612755'
TEAMS.PHI.NAME // 'sixers'

```

Tumne replied:
Also let's rename this file from `enums.js` to just `index.js` to follow your current file naming convention

Tumne commented on src/components/players/list/index.module.scss:
```diff
@@ -38,9 +38,9 @@
   }
 
   .imageLine {
+    border-bottom: 2px solid transparent;
```
move this after `margin-top: 4px;`

Commit:
[#53] Made pr fixes

Commit:
[#53] minor fixes

Commit:
[#53] minor fix

Commit:
Merge branch 'RAP-53__Team-select-refactor--A' into RAP-53__Team-select-refactor--B

Commit:
[#53] Pr fixes made

Commit:
[#53] minor fixes

Commit:
[#53] minor fix

Tumne committed:
[#53] Minor UI tweeks

Tumne committed:
Merge pull request #60 from kammanz/RAP-53__Team-select-refactor--B

[#53] Team select dropdown - B
